shop3/explicit-stack-search/explicit-search.lisp

```diff
@@ -243,43 +242,77 @@ List of analogical-replay tables -- optional
                (setf (mode state) 'expand-task)
                (stack-backtrack state)))
 
-          (expand-task
-           (let ((task (current-task state)))
-             (trace-print :tasks (get-task-name task) (world-state state)
-                          "~2%Depth ~s, trying task ~s"
-                          (depth state)
-                          (apply-substitution task (unifier state)))
-             (incf *expansions*)
-             (cond
-               ((primitivep (get-task-name task))
-                (setf (mode state) 'expand-primitive-task))
-               ((eql (get-task-name task) :loop)
-                (setf (mode state) 'unfold-looping-task))
-               (t ; original nonprimitive:
-                (setf (mode state) 'prepare-to-choose-method)))))
-
-          (unfold-looping-task
-           (when (> *verbose* 2) (format t "~%Starting to unfold the loop..."))
-           (if (unfold-loop-task domain state)
-               (progn
-                 (setf (mode state) 'test-for-done)
-                 (incf (depth state)))
-               ;; Else,
-               (with-slots (current-task depth world-state) state
-                 (when (> *verbose* 0) (format t "~%Could not unfold the loop successfully..."))
-                 (trace-print :tasks (get-task-name current-task) world-state
-                              "~2%Depth ~s, backtracking from task~%      task ~s"
-                              depth
-                              current-task)
-                 (stack-backtrack state))))
-
-          (expand-primitive-task
-           (if (expand-primitive-state state domain)
-               (progn
-                 (setf (mode state) 'test-for-done)
-                 (incf (depth state)))
-               (with-slots (current-task depth world-state) state
-                 (trace-print :tasks (get-task-name current-task) world-state
+        (expand-task
+         (let ((task (current-task state)))
+           (trace-print :tasks (get-task-name task) (world-state state)
+                        "~2%Depth ~s, trying task ~s"
+                        (depth state)
+                        (apply-substitution task (unifier state)))
+           (incf *expansions*)
+           (when (>= *verbose* 2)
+             (format t "~%Task name: ~s" (get-task-name task)))
+           (cond
+             ((primitivep (get-task-name task))
+              (setf (mode state) 'expand-primitive-task))
+             ((eql (get-task-name task) :loop)
```

Comment:
Is there a reason to use keywords here instead of just using the normal symbols?

I know this is a nit, but SHOP syntax is already full of inconsistencies about use of keywords versus normal symbols, and I would like to move towards consistency by killing all use of keywords inside method definitions, and reserve them only for `:op`, `:method`, etc.